py3status/modules/mega_sync.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""
+Display status of MEGA sync.
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module (default 10)
+    format: display format for this module (default "MEGA: {SyncState_0}")
+
+Format placeholders:
+    <column>_<ID> where <column> and <ID> are returned by 'mega-sync' command.
```

Comment:
Added, is that what you wanted?

Reply:
Yes. Still unclear to us who don't use `megasync`. I'll have to install this to understand what exactly the available placeholders are.... Thanks for adding. :+1:

Reply:
Yeah, I guess this is the case, you will run `mega-sync`, see the output including the values specific to your own installation of megasync and then you will see which values you want to see on py3status.

Here's mine, by the way, I have one row with ID=0, so everything is suffixed with `_0`:

```
❯ mega-sync 
ID LOCALPATH      REMOTEPATH     ActState   SyncState     SIZE  FILES   DIRS
 0 ~/MEGA         /              Active     Synced     4.45 GB   3275    678 
```

Reply:
Can you add more paths to sync? I think this calls for looping with `format_sync` or `format_path` instead of having you populating everything in one `format`. That explains why you used numbered prefixes. See `nvidia_smi`. It does basically what you want to do here.

Reply:
Exactly, yeah that's what I wanted, thanks

Reply:
Done now. I have one issues though, with this negative equals format doesn't seem to work...

What am I missing @lasers?

```
format_sync = "[\?!if=SyncState=Synced {SyncState}]"
```

I tried this:

```
format_sync = "[\?if=ID=0 {SyncState}]"
```

And it correctly shows me one of two entries, but if I put `!` before `if`, then I see two entries. Why?

```
format_sync = "[\?!if=ID=0 {SyncState}]"
```

Reply:
I'm not sure if it's good to keep keys as-is. Maybe lower the line before you make column keys.
```
ID LOCALPATH      REMOTEPATH     ActState   SyncState     SIZE  FILES   DIRS

... to... 

id, localpath, remotepath, actstate, syncstate, size, files, dirs
```
We don't have `[\?!if=`. We have `[\?if=!`. Try that.

Reply:
Keeping keys as-is is easy for copy-paste, requiring users to do post-processing will only lead for typos and wasted time... Let's keep them as is, we can always complain on MEGAcmd github that their headers are ugly and inconsistent if it really bothers you 😉 

> We don't have [\?!if=. We have [\?if=!. Try that.

Yeah, worked, thanks

Reply:
It does not bother me. What bother me is that we can break formats if we don't try and lowercase this... @tobes might disagree with keeping this as-is too.

We try to make our placeholders lowercase if possible. Otherwise, it's stupid to use `${REMOTEPATH}` because of their output... What if they change it later to `${RemotePath}` then we break lot of people's format just the same... so it's generally safe/better/more consistent if we keep or make all things lowercase.

Do you want to pop over there and ask them why their headers aren't inconsistent? :wink: